bodhi/server/consumers/masher.py

```diff
@@ -228,45 +250,52 @@ def _get_composes(self, msg):
 
             return [c.__json__(composer=True) for c in composes]
 
-    def work(self, msg):
-        """Begin the push process.
+    def consume(self, message: typing.Mapping):
+        """
+        Begin the push process.
 
         Here we organize & prioritize the updates, and fire off separate
         threads for each repo tag being mashed.
 
         If there are any security updates in the push, then those repositories
         will be executed before all others.
+
+        Note: This should be folded into __call__() when support for fedmsg is removed. It is
+        separate only because Masher.consume() receives a Munch object and fedora_messaging
+        receives a Message object and so this gives us a nice way to hand a Mapping from both.
+
```

Comment:
One thing to remember with fedora-messaging, is that duplicate messages may happen. Therefore I would recommend consuming classes to write explicitely in their documentation how they plan to handle duplicates. It may be as simple as "in case of duplicates, the task will be run again", but for more expensive tasks we could be looking at timestamps, use a locking system (and avoid deadlocks), etc.
It's one thing to keep in mind, and I think each consumers should document how they plan to handle it so we're sure it's not overlooked.